default/AI/ShipDesignAI.py

```diff
@@ -334,7 +335,12 @@ def _check_cache_for_consistency(self):
             print_error(e)
 
         corrupted = []
-        for designname in self.design_id_by_name:
+        # create a copy of the dict-keys so we can alter the dict
+        for designname in list(self.design_id_by_name.keys()):
+            # dropping invalid designs from cache
+            if self.design_id_by_name[designname] == EnumsAI.AIShipDesignTypes.SHIPDESIGN_INVALID:
+                self.design_id_by_name.pop(designname)
```

Comment:
seems to me that the pop here is entirely serving as a del, and simply using the latter would be more specific/clear